core/controllers/base_test.py

```diff
@@ -269,6 +270,23 @@ def test_i18n_keys(self):
                     self.log_line('- %s' % key)
                 self.log_line('')
 
+    def test_alphabetic_i18n_keys(self):
+        """Tests that the keys of all i18n json files are arranged in
+        alphabetical order."""
+        filenames = os.listdir(
+            os.path.join(os.getcwd(), self.get_static_asset_filepath(),
+                         'assets', 'i18n'))
+        for filename in filenames:
+            with open(os.path.join(os.getcwd(), 'assets', 'i18n', filename),
+                      mode='r') as f:
+                for line in f:
+                    line = line.replace("{", "")
+                    line = line.replace("}", "")
+                    line = line.partition(':')
+                    line = line[0].split()
+                    ord_line = sorted(line)
```

Comment:
Hello @wxyxinyu <https://github.com/wxyxinyu> , I've pushed some new
changes. Could you please review it??

On Tue, Sep 6, 2016 at 7:39 AM, Xinyu Wu <notifications@github.com> wrote:

> In core/controllers/base_test.py
> <https://github.com/oppia/oppia/pull/2426#discussion_r77579754>:
>
> > @@ -269,6 +270,23 @@ def test_i18n_keys(self):
> >                      self.log_line('- %s' % key)
> >                  self.log_line('')
> >
> > +    def test_alphabetic_i18n_keys(self):
> > +        """Tests that the keys of all i18n json files are arranged in
> > +        alphabetical order."""
> > +        filenames = os.listdir(
> > +            os.path.join(os.getcwd(), self.get_static_asset_filepath(),
> > +                         'assets', 'i18n'))
> > +        for filename in filenames:
> > +            with open(os.path.join(os.getcwd(), 'assets', 'i18n', filename),
> > +                      mode='r') as f:
> > +                for line in f:
> > +                    line = line.replace("{", "")
> > +                    line = line.replace("}", "")
> > +                    line = line.partition(':')
> > +                    line = line[0].split()
> > +                    ord_line = sorted(line)
>
> This seems to be sorting the line itself (i.e., the tuple containing the
> key, ':', and the value) rather than checking the sortedness across lines.
> You should be checking that line is lexicographically after the previous
> line, and not checking sorting within the line.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/2426/files/83e803b85c383fdfddfbd74881275eb335bf5b05#r77579754>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANMtlR52px7aKktFcSMkmwieM1q0v5bgks5qnQq9gaJpZM4JsT8G>
> .
>